nevergrad/benchmark/xpbase.py

```diff
@@ -57,15 +59,20 @@ def __init__(
         self.optimizer = optimizer
         self.budget = budget
         self.num_workers = num_workers
-        self.executor = execution.MockedTimedExecutor(batch_mode)
+        ng_cpus_per_task = int(os.getenv("NG_CPUS_PER_TASK", "1"))
+        if ng_cpus_per_task > 1:
+            self.executor: tp.Any = futures.ThreadPoolExecutor(max_workers=ng_cpus_per_task)
+            assert not batch_mode
+        else:
+            self.executor = execution.MockedTimedExecutor(batch_mode)
 
     @property
     def name(self) -> str:
         return self.optimizer if isinstance(self.optimizer, str) else repr(self.optimizer)
 
     @property
     def batch_mode(self) -> bool:
-        return self.executor.batch_mode
+        return hasattr(self.executor, "batch_mode") and self.executor.batch_mode
```

Comment:
```suggestion
        return isinstance(self.executor, execution.MockedTimedExecutor) and self.executor.batch_mode
```

Reply:
checking hasattr is not safe, if we update the code that could break without notice, it's much safer to test against a class